File: benchmarks/3n-plus-1/Main.java
public class Main {

    public static long next(long n) {
        if (n % 2 == 0) {
            return n / 2;
        } else {
            return 3 * n + 1;
        }
    }

    public static void main(String[] args) {
        long before = System.nanoTime();

        long n = 837799L;
        while (n != 1L) {
            n = next(n);
        }

        long after = System.nanoTime();

        System.out.println(after - before);
    }

}

File: benchmarks/3n-plus-1/main.rs
use std::time::{SystemTime};

fn next(n: u64) -> u64 {
    if n % 2 == 0 {
        n / 2
    } else {
        3 * n + 1
    }
}

fn main() {
    let before = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).unwrap();

    let mut n = 837799;
    while n != 1 {
        n = next(n);
    }

    let after = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).unwrap();
    println!("{}", after.as_nanos() - before.as_nanos());
}